lib/gossip/ping-sender.js

```diff
@@ -57,40 +54,41 @@ PingSender.prototype.onPing = function onPing(err, res) {
         res: res
     });
     this.ring.membership.update(res.changes);
-    this.callback(err, res);
-    this.callback = null;
+    callback(err, res);
     return;
 };
 
-PingSender.prototype.send = function send() {
+PingSender.prototype.sendChanges = function sendChanges(changes, callback) {
     var self = this;
 
-    self.ring.dissemination.issueAsSender(function issue(changes, onIssue) {
-        self.gossipLogger.info('ringpop ping send', {
-            local: self.ring.whoami(),
-            member: self.address,
-            changes: changes
-        });
+    self.ring.client.protocolPing({
+        host: self.address,
+        retryLimit: self.ring.config.get('tchannelRetryLimit'),
+        timeout: self.ring.pingTimeout
+    }, {
+        checksum: self.ring.membership.checksum,
+        changes: changes,
+        source: self.ring.whoami(),
+        sourceIncarnationNumber: self.ring.membership.getIncarnationNumber()
+    }, callback);
+};
 
-        self.ring.client.protocolPing({
-            host: self.address,
-            retryLimit: self.ring.config.get('tchannelRetryLimit'),
-            timeout: self.ring.pingTimeout
-        }, {
-            checksum: self.ring.membership.checksum,
-            changes: changes,
-            source: self.ring.whoami(),
-            sourceIncarnationNumber: self.ring.membership.getIncarnationNumber()
-        }, function onPing(err, res) {
+PingSender.prototype.send = function send(callback) {
+    var self = this;
+
+    self.ring.dissemination.issueAsSender(function issues(changes, onIssue) {
+        self.sendChanges(changes, function onPing(err, res) {
```

Comment:
You're right, I'll add the logging again (to the `sendChanges`-method) and change the onPing as well.